x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log_adapter/event_log_adapter.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEventLogService } from '../../../../../../event_log/server';
+import {
+  FindBulkExecutionLogArgs,
+  FindExecutionLogArgs,
+  IRuleExecutionLogClient,
+  LogExecutionMetricsArgs,
+  LogStatusChangeArgs,
+  UpdateExecutionLogArgs,
+} from '../types';
+import { EventLogClient } from './event_log_client';
+
+export class EventLogAdapter implements IRuleExecutionLogClient {
```

Comment:
Curious what do you think about error handling in IRuleExecutionLogClient implementations.

1. are we gonna throw and delegate it to rule executors?
2. are we gonna swallow exceptions inside and log them via Kibana logger?
3. or catch exceptions inside, write meaningful logs via Kibana logger and re-throw as custom exceptions?
4. or smth else?

Could be addressed separately.

Reply:
Just to be clear, I believe the base event log `logEvent()` function will never throw an error.  It will log a message if the event param is not valid.  And is otherwise synchronous, so stack traces should show the caller.  So we shouldn't have to do anything about the calls that are adding event log docs.

However, presumably `find()`, `findBulk()`, etc, when implemented could, so it would be good to have a nice story here for errors.  I'd suggest at least logging a message when caught, but whether you re-throw or throw a different error depends on the caller (and I don't know who would be calling these).

Reply:
Thanks for the suggestion 👍  We'll cover error handling in the follow-up PR.